depthai_nodes/parsing_neural_network.py

```diff
@@ -0,0 +1,130 @@
+from pathlib import Path
+
+import depthai as dai
+
+from .ml.parsers import BaseParser, ParserGenerator
+
+
+class ParsingNeuralNetwork(dai.node.ThreadedHostNode):
+    Propeties = dai.node.NeuralNetwork.Properties
+
+    def __init__(self, *args, **kwargs) -> None:
+        super().__init__(*args, **kwargs)
+        self._pipeline = self.getParentPipeline()
+        self._nn = self._pipeline.create(dai.node.NeuralNetwork)
+        self._parsers: dict[int, BaseParser] = {}
+
+    def build(
+        self, input: dai.Node.Output, nnArchive: dai.NNArchive
+    ) -> "ParsingNeuralNetwork":
+        self._nn_archive = nnArchive
+        self._nn.build(input, nnArchive)
+        self._updateParsers(nnArchive)
+        return self
+
+    def _updateParsers(self, nnArchive: dai.NNArchive) -> None:
+        self._removeOldParserNodes()
+        self._parsers = self._getParserNodes(nnArchive)
+
+    def _removeOldParserNodes(self) -> None:
+        for parser in self._parsers.values():
+            self._pipeline.remove(parser)
+
+    def _getParserNodes(self, nnArchive: dai.NNArchive) -> dict[int, BaseParser]:
+        parser_generator = self._pipeline.create(ParserGenerator)
+        parsers = parser_generator.build(nnArchive)
+        for parser in parsers.values():
+            self._nn.out.link(parser.input)
```

Comment:
Can we more specifically define this? (`_nn.out["output_name"].link`)?

Reply:
I am not sure if I understand what you mean. Right now the only binding in DepthAI is this one:
`    def out(self) -> depthai.Node.Output: ...`

Reply:
I believe in v2 at least there was plan to support linking specific outputs from NN nodes. So if your NN had 3 outputs ("cls", "seg", "det"), you could link like:
```
nn.out["cls"].link(cls_parser.input)
nn.out["seg"].link(seg_parser.input)
```
This requires less data to be sent between the nodes. We should check with DAI team if this is in place. If not it's OK as is, but might be worth leaving a TODO here?

Reply:
Okay, I've left a TODO.